src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RecordAdjournmentDetailsMidEventHandler.java

```diff
@@ -34,15 +39,32 @@ public PreSubmitCallbackResponse<AsylumCase> handle(PreSubmitCallbackStage callb
         if (!canHandle(callbackStage, callback)) {
             throw new IllegalStateException("Cannot handle callback");
         }
-
         AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
         asylumCase.read(HEARING_CHANNEL, DynamicList.class)
-            .ifPresent(hearingChannel -> asylumCase.write(NEXT_HEARING_FORMAT, hearingChannel));
+                .ifPresent(hearingChannel -> {
+                    Optional<DynamicList> nextHearingFormat = asylumCase.read(NEXT_HEARING_FORMAT, DynamicList.class);
+                    nextHearingFormat.ifPresent(it -> it.setValue(hearingChannel.getValue()));
+                    asylumCase.write(NEXT_HEARING_FORMAT, nextHearingFormat.get());
+                });
+
         asylumCase.read(LIST_CASE_HEARING_LENGTH, String.class)
-            .ifPresent(hearingLength -> asylumCase.write(NEXT_HEARING_DURATION, hearingLength));
-        asylumCase.read(LIST_CASE_HEARING_CENTRE, HearingCentre.class)
-            .ifPresent(hearingCentre -> asylumCase.write(NEXT_HEARING_LOCATION, hearingCentre));
+                .ifPresent(hearingLength -> asylumCase.write(NEXT_HEARING_DURATION, hearingLength));
+        asylumCase.read(HEARING_CENTRE, HearingCentre.class)
+                .ifPresent(hearingCentre -> asylumCase.write(NEXT_HEARING_LOCATION, hearingCentre));
+        return responseCheckingHearingRange(asylumCase);
+    }
 
-        return new PreSubmitCallbackResponse<>(asylumCase);
+    private PreSubmitCallbackResponse<AsylumCase> responseCheckingHearingRange(AsylumCase asylumCase) {
```

Comment:
minor one but could use a more meaningful name for this method like validateHearingDateRange